### ISSUE TITLE
Migration lock was never released or currently a migration is running.

### ISSUE THREAD

**Issue by ethwillupto10000** [TryGhost/Ghost#19131]
### Issue Summary

Debug Information:
    OS: Ubuntu, v22.04.3 LTS
    Node Version: v18.18.0
    Ghost Version: 5.72.0
    Ghost-CLI Version: 1.25.3
    Environment: production
    Command: 'ghost setup'

### Steps to Reproduce

✔ Setting up "ghost" system user
ℹ Setting up "ghost" mysql user [skipped]
Nginx is not installed. Skipping Nginx setup.
ℹ Setting up Nginx [skipped]
Nginx setup task was skipped, skipping SSL setup
ℹ Setting up SSL [skipped]
Systemd service has already been set up. Skipping Systemd setup
ℹ Setting up Systemd [skipped]
+ sudo systemctl is-active ghost_zee-com
? Do you want to start Ghost? Yes
+ sudo systemctl start ghost_zee-com
+ sudo systemctl stop ghost_zee-com
✖ Starting Ghost
One or more errors occurred.

1) GhostError

Message: Ghost was able to start, but errored during boot with: Migration lock was never released or currently a migration is running.
Help: If you are sure no migration is running, check your data and if your database is in a broken state, you could run `yarn knex-migrator rollback`.
Suggestion: journalctl -u ghost_zee-com -n 50

Debug Information:
    OS: Ubuntu, v22.04.3 LTS
    Node Version: v18.18.0
    Ghost Version: 5.72.0
    Ghost-CLI Version: 1.25.3
    Environment: production
    Command: 'ghost setup'

### Ghost Version

5.72.0

### Node.js Version

18.18.0

### How did you install Ghost?

ubuntu 22.04

### Database type

MySQL 8

### Relevant log / error output

```shell
{
   message: {
     statusCode: 500,
     errorType: 'MigrationsAreLockedError',
     level: 'critical',
     message: 'Ghost was able to start, but errored during boot with: Migration lock was never released or currently a migration is running.',
     id: 500,
     help: 'If you are sure no migration is running, check your data and if your database is in a broken state, you could run `yarn knex-migrator rollback`.',
     name: 'MigrationsAreLockedError',
     code: null,
     property: null,
     redirect: null,
     hideStack: false
   }
 }
Node.js v18.18.0
```

### Code of Conduct

- [X] I agree to be friendly and polite to people in this repository

**Comment by ethwillupto10000** [comment]
When i try to install another ghost instance on the same machine, error occur....
Old instance was on importing data from remote server(Migrating data to ghost)

**Comment by aztechian** [comment]
This has bricked my installation of ghost using the Kubernetes helm charts. When I try to run the suggested rollback command, I get

```
I have no name!@ghost-unmigrator:/opt/bitnami/ghost/current$ NODE_ENV=production yarn knex-migrator rollback 
yarn run v1.22.19
warning Skipping preferred cache folder "/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-1001".
warning ghost@5.74.1: The engine "cli" appears to be invalid.
$ /opt/bitnami/ghost/versions/5.74.1/node_modules/.bin/knex-migrator rollback
warning Cannot find a suitable global folder. Tried these: "/usr/local, /.yarn"
[2023-11-25 22:01:53] ERROR connect ECONNREFUSED 127.0.0.1:3306
[2023-11-25 22:01:53] INFO Unknown database error
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
I have no name!@ghost-unmigrator:/opt/bitnami/ghost/current$ 
```

and I can't find any documentation on how to get the migrator to pick up the database settings from ENV vars. They all exist on the pod for running the usuall ghost app.

```
$ env | grep DATABASE | sort
GHOST_DATABASE_HOST=ghost-mysql
GHOST_DATABASE_NAME=bitnami_ghost
GHOST_DATABASE_PASSWORD=<REDACTED>
GHOST_DATABASE_PORT_NUMBER=3306
GHOST_DATABASE_USER=bn_ghost
```

I tried deleting the single record from the `migrations_lock` table, and restarting the app but still got the same message that a migration is running.
I believe the problem initially started with the `19.6.11` version of the helm chart (on Wed 11/22) which corresponds to the `5.74.1` app version.

**Comment by daniellockyer** [comment]
@aztechian You need to run knex-migrator from the directory containing Ghost code 🙂 

@ethwillupto10000 Do you know what caused Ghost to be killed/error when it was running migrations? The manual way to remove the migration lock is to set `locked=0` in the `migrations_lock` table, but only do that if you're sure there are no other instances running.

5.74.1 contained no Ghost backend changes, so it's unlikely to be due to that.

**Comment by aztechian** [comment]
> @aztechian You need to run knex-migrator from the directory containing Ghost code 🙂

I did get this resolved once I tried a `knex-migrator migrate` command that mentioned setting the migration_lock field `locked=0`. I still didn't have a record recreated in that table, so created one with `locked=0`. Then ran `migrate` manually, which did some changes. After that the health said OK, and the app could start.
Thanks for the pointers @daniellockyer. Bitnami's environment setup is kinda weird - it took me way to long to figure out where the real code directory is, and I had to symlink `/opt/bitnami/ghost/current/MigratorConfig.js -> /opt/bitnami/ghost/MigratorConfig.js` to get migrator to run at all.
I'm back up and running.
